Neuro/Impostor/ImpostorHandler.cs

```diff
@@ -0,0 +1,243 @@
+                }
+            }
+        }
+        ClosestVent = closest;
+
+        if (ClosestVent == null)
+        {
+            DirectionToNearestVent = Vector2.zero;
+        }
+        else
+        {
+            DirectionToNearestVent = (ClosestVent.transform.position - PlayerControl.LocalPlayer.transform.position).normalized;
+        }
+        
+    }
+
+    [HideFromIl2Cpp]
+    private void UpdateNearbyDoors()
+    {
+        NearbyDoors.Clear();
+        foreach (PlainDoor door in ShipStatus.Instance.AllDoors)
+        {
+            float distance = Vector2.Distance(door.transform.position, PlayerControl.LocalPlayer.transform.position);
+            if (distance < 10f)
+            {
+                NearbyDoors.Add(door);
+            }
+        }
+    }
+
+    /** ----- Stubs below here ----- **/
+
+    // example of kill functionality
+    private void GetOrKillTarget()
+    {
+        if (PlayerControl.LocalPlayer.killTimer > 0f) return;
+
+        if (killTarget == null && !PlayerControl.LocalPlayer.inVent)
+        {
+            PlayerControl potentialKillTarget = null;
+
+            foreach (KeyValuePair<PlayerControl, LastSeenPlayer> player in NeuroPlugin.Instance.Vision.GetPlayerLocations())
+            {
+                // ignore ourselves, dead players, and other impostors
+                if (PlayerControl.LocalPlayer == player.Key) continue;
+                if (player.Key.Data.IsDead) continue;
+                if (player.Key.Data.RoleType == AmongUs.GameOptions.RoleTypes.Impostor) continue;
+
+                if (player.Value.time > Time.timeSinceLevelLoad - (2 * Time.fixedDeltaTime))
+                {
+                    // if there are multiple players in view, avoid trying to kill so we dont give ourselves up
+                    if (potentialKillTarget != null)
+                    {
+                        potentialKillTarget = null;
+                        break;
+                    }
+                    potentialKillTarget = player.Key;
+                }
+            }
+            // if the target ends up being the only target in the room, mark them a kill target and pathfind to them
+            if (potentialKillTarget != null)
+            {
+                killTarget = potentialKillTarget;
+            }
+        }
+        else
+        {
+            if (!goingForKill && HudManager.Instance.KillButton.currentTarget != null)
+            {
+                goingForKill = true;
+                PlayerControl.LocalPlayer.StartCoroutine(MurderTarget());
+            }
+        }
+    }
+
+    [HideFromIl2Cpp]
+    private IEnumerator MurderTarget()
+    {
+        // wait a random amount of time so we dont just instantly kill them the second we are in range
+        yield return new WaitForSeconds(UnityEngine.Random.RandomRange(0.2f, 0.6f));
+        // this actually works in practice area which is pretty nice
+        PlayerControl.LocalPlayer.MurderPlayer(killTarget);
```

Comment:
What if the player moves? You need to pathfind to them

Reply:
I had implemented player and vent pathfinding previously however have since removed them since I didn't think we were using these functions anymore. However if we are, I can reimplement the player and vent pathfinding.